### PR TITLE
feat: separate public project list as a menu item

### PR DESCRIPTION
**Pull request by sherotree**
## What type of PR is this?

- [x] Feature
- [ ] Bugfix
- [ ] Test
- [ ] Documentation
- [ ] Refactor
- [ ] Style
- [ ] Chore

## Which issue(s) this PR fixes:

Fixes #

## Does this PR introduce a user interface change?

- [x] Yes(screenshot is required)
- [ ] No
image

### CODE REVIEW

File: shell/app/menus/workBench.tsx [hunk 1]
```diff
@@ -95,6 +96,11 @@ export const getWorkBenchMenu = () => {
       text: i18n.t('publisher:joined publisher'),
       show: orgPerm.workBench.publisher.read.pass,
     },
+    {
+      href: goTo.resolve.workBenchPublicProjects(),
+      icon: <BookOne />,
+      text: i18n.t('public project'),
```
Comment (daskyrk) [inline shell/app/menus/workBench.tsx:102]: missing permission config

File: shell/app/modules/project/types/project.d.ts [hunk 2]
```diff
@@ -21,6 +21,11 @@ declare namespace PROJECT {
     TEST: T
   }
 
+  interface ProjectListTypeParams {
```
Comment (McDaddy) [inline shell/app/modules/project/types/project.d.ts:24]: If the interface is used only in a single file, then I think there is no need to define it in global namespace. U can just define it in the specific file

File: shell/app/modules/workBench/pages/projects/project-list-protocol.tsx [hunk 3]
```diff
@@ -75,6 +76,10 @@ export const ProjectList = () => {
   );
 };
 
+export const ProjectListMy = () => <ProjectList scenarioKey='project-list-my'
+  scenarioType='project-list-my' key='project-list-my' />
+export const ProjectListAll = () => <ProjectList scenarioKey='project-list-all'
+  scenarioType='project-list-all' key='project-list-all' />
```
Comment (McDaddy) [inline shell/app/modules/workBench/pages/projects/project-list-protocol.tsx:82]: How about to check route path equals `projects ` or `public-projects` to distinguish these two components, in the future, it's better to export route level Comp by default export

File: shell/app/modules/workBench/pages/projects/project-list-protocol.tsx [hunk 4]
```diff
@@ -16,13 +16,33 @@ import DiceConfigPage from 'config-page/index';
...
+
+  React.useEffect(() => {
+    if (get(currentRoute, 'relativePath') === 'public-projects') {
+      update({
+        scenarioKey: 'project-list-all',
+        scenarioType: 'project-list-all',
+      });
+    }
+
+    if (get(currentRoute, 'relativePath') === 'projects') {
```
Comment (McDaddy) [inline shell/app/modules/workBench/pages/projects/project-list-protocol.tsx:39]: no need if again, `else` is enough

File: shell/app/modules/workBench/pages/projects/project-list-protocol.tsx [hunk 5]
```diff
@@ -16,13 +16,33 @@ import DiceConfigPage from 'config-page/index';
...
 import ApplyUnblockModal, { IMetaData } from 'workBench/pages/projects/apply-unblock-modal';
+import routeInfoStore from 'app/common/stores/route';
 
-export const ProjectList = () => {
+const ProjectList = () => {
   const [state, updater, update] = useUpdate({
     visible: false,
     metaData: {} as IMetaData,
+    scenarioKey: '',
+    scenarioType: '',
```
Comment (McDaddy) [inline shell/app/modules/workBench/pages/projects/project-list-protocol.tsx:26]: actually, u can initialize state here and remove useEffect, the problem of using `useEffect` is the first time page render these two state would be empty.
```
scenarioKey: currentRoute.relativePath === 'public-projects' ? 'project-list-all' : 'project-list-my',
scenarioType: currentRoute.relativePath === 'public-projects' ? 'project-list-all' : 'project-list-my',
````

**Review by McDaddy — APPROVED** [review]

**Pull request merged**